### PR TITLE
udev: tiny cleanups

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/udev/udevd.c [hunk 1]
```diff
@@ -186,18 +186,7 @@ static void worker_free(struct worker *worker) {
...
-        Iterator i;
-
-        assert(manager);
-
-        HASHMAP_FOREACH(worker, manager->workers, i)
-                worker_free(worker);
-
-        manager->workers = hashmap_free(manager->workers);
-}
+DEFINE_PRIVATE_HASH_OPS_WITH_VALUE_DESTRUCTOR(worker_hash_op, void, trivial_hash_func, trivial_compare_func, struct worker, worker_free);
```
Comment (keszybz) [inline src/udev/udevd.c:190]: Nice.

**Comment by keszybz** [comment]
```
325/726 test-mountpoint-util                    FAIL     0.01 s (killed by signal 6 SIGABRT)
--- command ---
PATH='/build/build:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin' SYSTEMD_LANGUAGE_FALLBACK_MAP='/build/src/locale/language-fallback-map' SYSTEMD_KBD_MODEL_MAP='/build/src/locale/kbd-model-map' /build/build/test-mountpoint-util
--- stderr ---
Found container virtualization container-other.
mnt id of /dev/mqueue is 110
mnt id of /dev/pts is 94
mnt id of /sys/fs/cgroup/blkio is 100
mnt id of /sys/fs/cgroup/cpu is 98
mnt id of /sys/fs/cgroup/pids is 108
mnt id of /sys/kernel/security is 57
mnt id of /sys is 95
mnt id of /dev is 93
mnt id of /dev/console is 56
mnt id of /sys/fs/bpf is 60
mnt id of /sys/fs/cgroup/memory is 101
mnt id of /sys/fs/cgroup/cpuset is 97
mnt id of /sys/fs/cgroup/net_cls is 104
mnt id of /proc/sys/fs/binfmt_misc is 61
mnt id of /dev/shm is 115
mnt id of /sys/kernel/debug is 62
mnt id of /build/build/test/tmpfs is 111
Assertion 'path_equal_ptr(hashmap_get(h, INT_TO_PTR(mnt_id2)), p)' failed at ../src/test/test-mountpoint-util.c:78, function test_mnt_id(). Aborting.
-------
```
Hmm.

**Comment by keszybz** [comment]
I posted https://github.com/systemd/systemd/pull/11197 which puts more debug info in test-mountpoint-util.

**Comment by evverx** [comment]
> 325/726 test-mountpoint-util                    FAIL     0.01 s (killed by signal 6 SIGABRT)

It could have had something to do with overlayfs. In https://github.com/systemd/systemd/issues/11092#issuecomment-445482286 @poettering referred to a bug causing overlayfs to report the wrong mnt_id. It could have been a variation of it.

**Comment by keszybz** [comment]
`bionic-amd64`: `AssertionError: Regex didn't match: b'kernel:.*[cC]ommand line:' not found in b'`

**Comment by poettering** [comment]
hmm, any chance you can rebase/repush this PR, so that the CI has another chance to pass?

**Comment by poettering** [comment]
> `bionic-amd64`: `AssertionError: Regex didn't match: b'kernel:.*[cC]ommand line:' not found in b'`

See #11104

**Comment by keszybz** [comment]
I'll just retrigger the bionic builds.

**Comment by keszybz** [comment]
Something seems to be busted. Failure on arm64 seems harmless. But bionic-amd64:
```
[  OK  ] Started Create Static Device Nodes in /dev.
[  OK  ] Reached target Local File Systems (Pre).
         Starting udev Kernel Device Manager...
[  OK  ] Started udev Kernel Device Manager.
[*     ] (1 of 3) A start job is running for…8c40-5825c5bf44ef (27s / 1min 36s)
^M[**    ] (1 of 3) A start job is running for…8c40-5825c5bf44ef (28s / 1min 36s)
^M[***   ] (1 of 3) A start job is running for…8c40-5825c5bf44ef (28s / 1min 36s)
^M[ ***  ] (2 of 3) A start job is running for…v/mapper/varcrypt (29s / no limit)
^M[  *** ] (2 of 3) A start job is running for…v/mapper/varcrypt (29s / no limit)
^M[   ***] (2 of 3) A start job is running for…v/mapper/varcrypt (30s / no limit)
^M[    **] (3 of 3) A start job is running for /dev/ttyS0 (30s / 1min 35s)
^M[     *] (3 of 3) A start job is running for /dev/ttyS0 (31s / 1min 35s)
^M[    **] (3 of 3) A start job is running for /dev/ttyS0 (31s / 1min 35s)
^M[   ***] (1 of 3) A start job is running for…8c40-5825c5bf44ef (32s / 1min 36s)
^M[  *** ] (1 of 3) A start job is running for…8c40-5825c5bf44ef (32s / 1min 36s)
^M[ ***  ] (1 of 3) A start job is running for…8c40-5825c5bf44ef (33s / 1min 36s)
...
^M[***   ] (2 of 3) A start job is running for…pper/varcrypt (1min 3s / no limit)
^M[ ***  ] (2 of 3) A start job is running for…pper/varcrypt (1min 4s / no limit)
^M[  *** ] (3 of 3) A start job is running for /dev/ttyS0 (1min 5s / 1min 35s)
^M[  OK  ] Found device /dev/ttyS0.
[  OK  ] Found device QEMU_HARDDISK____ 2.
[   ***] A start job is running for /dev/mapper/varcrypt (1s / no limit)
^M         Starting Cryptography Setup for varcrypt...
[    **] (1 of 2) A start job is running for…Setup for varcrypt (2s / no limit)
^M[     *] (2 of 2) A start job is running for…ev/mapper/varcrypt (2s / no limit)
^M[    **] (2 of 2) A start job is running for…ev/mapper/varcrypt (3s / no limit)
...
^M[  OK  ] Found device /dev/mapper/varcrypt.
[  OK  ] Started Cryptography Setup for varcrypt.
[  OK  ] Reached target Local Encrypted Volumes.
         Starting File System Check on /dev/mapper/varcrypt...
[  OK  ] Started File System Check on /dev/mapper/varcrypt.
         Mounting /var...
[FAILED] Failed to mount /var.
See 'systemctl status var.mount' for details.
[DEPEND] Dependency failed for Local File Systems.
[DEPEND] Dependency failed for Upda…MP about System Boot/Shutdown.
[DEPEND] Dependency failed for Upda…about System Runlevel Changes.
[DEPEND] Dependency failed for Flus…Journal to Persistent Storage.
[DEPEND] Dependency failed for Load/Save Random Seed.
[DEPEND] Dependency failed for Load…itch Status /dev/rfkill Watch.
[  OK  ] Stopped Dispatch Password …ts to Console Directory Watch.
[  OK  ] Stopped Forward Password R…uests to Wall Directory Watch.
[  OK  ] Reached target Timers.
[  OK  ] Reached target Sockets.
[  OK  ] Reached target Login Prompts.
[  OK  ] Started Emergency Shell.
[  OK  ] Reached target Emergency Mode.
         Starting Create Volatile Files and Directories...
         Starting Commit a transient machine-id on disk...
[  OK  ] Started Commit a transient machine-id on disk.
[  OK  ] Started Create Volatile Files and Directories.
You are in emergency mode.
```
bionic-i386 also fails like that.

**Comment by mbiebl** [comment]
@keszybz I can reproduce this failure locally.. This started to happen here once #10268 was merged

**Comment by evverx** [comment]
TEST-02-CRYPTSETUP is flaky (https://github.com/systemd/systemd/issues/10093). It has just been turned off on Ubuntu CI: https://salsa.debian.org/systemd-team/systemd/commit/2c194b155b1e97e.

**Comment by yuwata** [comment]
Still I am not sure what causes the CI failures. So, I've just rebased this. I hope the failures are fixed by #11178.

**Comment by mbiebl** [comment]
I can reproduce the failure on amd64 rather reliably.
Reverting 89f9752ea08f516b5d77f8e577bb772073c70c01 and fcfb1f775ed0e9d282607bb118ba788b98952855 fixes this test.

**Comment by evverx** [comment]
@mbiebl when a test is flaky bugs causing it to fail tend to pile up and eventually nobody knows what exactly is wrong. Could you open an issue if you think those two commits have anything to do with the failure?

**Comment by keszybz** [comment]
I'll retrigger builds now that #11212 has been merged.

**Comment by keszybz** [comment]
bionic-i386 failed with "FAILED TESTS:  test/TEST-17-UDEV-WANTS". I'll restart it.

**Comment by evverx** [comment]
By the way, test/TEST-17-UDEV-WANTS is as flaky as TEST-02-CRYPTSETUP: https://github.com/systemd/systemd/issues/10274. It isn't skipped yet on Ubuntu CI but I hope it's just a matter of time.

**Comment by mrc0mmand** [comment]
CentOS CI is dying due to a copr outage, will retrigger it once it's working again.

**Comment by yuwata** [comment]
Rebased, and re-added tentatively dropped commit (the last one).

**Comment by keszybz** [comment]
Still LGTM.

**Pull request merged**